spyder/plugins/updatemanager/tests/test_update_manager.py

```diff
@@ -76,57 +78,61 @@ def test_updates(qtbot, mocker, caplog, version, channel):
-    version = parse(version)
-    mocker.patch.object(workers, "CURRENT_VERSION", new=version)
+    mocker.patch.object(workers, "CURRENT_VERSION", new=parse(version))
 
-    release = parse("6.1.0a2")
     worker = WorkerUpdate(stable_only)
+    release = parse(release)
+
     worker._check_update_available(release)
 
-    if stable_only:
-        if version.is_prerelease:
-            # We shouldn't offer an update from 6.1.0a1 to 6.1.0a2 when
-            # stable_only is True.
-            assert worker.asset_info is None
-        else:
-            # This should give the latest 6.0.x stable version
-            assert not worker.asset_info["version"].is_prerelease
+    if expected is not None:
+        expected = parse(expected)
+        assert worker.asset_info["version"] == expected
     else:
-        # For both pre and stable releases we should offer to update to the
-        # version used above.
-        assert worker.asset_info["version"] == release
+        assert worker.asset_info is None
 
 
-@pytest.mark.parametrize("version", ["4.0.0", "6.0.0"])
-@pytest.mark.parametrize("release", [None, "6.0.1", "6.100.0"])
-def test_update_no_asset(qtbot, mocker, version, release):
+def test_update_no_asset(qtbot, mocker):
     """Test update availability when asset is not available"""
-    mocker.patch.object(workers, "CURRENT_VERSION", new=parse(version))
+    mocker.patch.object(workers, "CURRENT_VERSION", new=parse("6.0.0b2"))
+    mocker.patch.object(workers, "is_conda_based_app", return_value=True)
 
-    release = parse(release) if release else None
-    worker = WorkerUpdate(True)
-    worker._check_update_available(release)
+    worker = WorkerUpdate(False)
+
+    # 6.0.0b3 does not have correct minor/micro update asset
+    worker._check_update_available(parse("6.0.0b3"))
+    assert worker.asset_info is None
 
-    # For both values of version, there should be an update available
-    # However, the available version should be 6.0.1, since there is
-    # no asset for 6.100.0
+    # 6.0.0rc1 is first release to have correct minor/micro update asset
+    worker._check_update_available(parse("6.0.0rc1"))
     assert worker.asset_info is not None
-    assert worker.asset_info["version"] >= parse("6.0.1")
+    assert worker.asset_info["version"] == parse("6.0.0rc1")
 
 
 @pytest.mark.parametrize(
     "app,version,release,update_type",
     [
-        (True, "6.0.0", "6.0.1", UpdateType.Micro),
-        (True, "6.0.0", "6.1.0a1", UpdateType.Minor),
-        (True, "5.0.0", "6.0.0", UpdateType.Major),
-        (False, "6.0.0", "6.0.1", UpdateType.Major),
-        (False, "6.0.0", "6.1.0a1", UpdateType.Major),
-        (False, "5.0.0", "6.0.0", UpdateType.Major)
+        (True, "6.0.0", "6.0.7", UpdateType.Micro),
+        (True, "6.0.0", "6.1.0a3", UpdateType.Minor),
+        (True, "5.0.0", "6.0.7", UpdateType.Major),
+        (False, "6.0.0", "6.0.7", UpdateType.Major),
+        (False, "6.0.0", "6.1.0a3", UpdateType.Major),
```

Comment:
Why updating from 6.0.0 to 6.1.0a3 is detected as a major update when not using our installers? This should be minor, shouldn't it?

Reply:
If 6.0.0 is installed *without* our installer, then *any* update will be classified internally as a major update so that the installer will be downloaded. Minor and micro updates will result in downloading the lock file and only apply to existing installations that used our installer.

Reply:
Got it, thanks for the clarification.